server/services/formService.js

```diff
@@ -80,33 +80,62 @@ module.exports = function createSomeService(formClient) {
     }
   }
 
-  async function referToSecurityIfRiskAssessed(bookingId, userId, socProfile) {
+  async function referToSecurityIfRiskAssessed(bookingId, userId, socProfile, currentStatus) {
     if (socProfile.transferToSecurity) {
-      try {
-        const result = await formClient.referToSecurity(bookingId, null, Status.SECURITY_AUTO.name)
-        if (result.rowCount === 0) {
-          // May be no record in db yet - ensure this is the case and insert it
-          const current = await getCategorisationRecord(bookingId)
-          if (current.form_response) {
-            throw new Error(`Invalid state, booking id ${bookingId}, status ${current.status}`)
-          }
-          await formClient.update(null, '{}', bookingId, userId, Status.STARTED.name, userId, null)
+      let status
+      if (!currentStatus) {
+        // No record in db yet - ensure we have an initial record
+        await formClient.update(null, '{}', bookingId, userId, Status.STARTED.name, userId, null)
+        status = Status.STARTED
+      } else {
+        status = Status[currentStatus]
+      }
+      if (Status.SECURITY_AUTO.previous.includes(status) && currentStatus !== Status.SECURITY_AUTO) {
+        try {
           await formClient.referToSecurity(bookingId, null, Status.SECURITY_AUTO.name)
+        } catch (error) {
+          logger.error(error)
+          throw error
         }
-      } catch (error) {
-        logger.error(error)
+      } else {
+        logger.warn(`Cannot transition from status ${status && status.name} to SECURITY_AUTO, bookingId=${bookingId}`)
       }
     }
     return {}
   }
 
   async function referToSecurityIfRequested(bookingId, userId, updatedFormObject) {
     if (updatedFormObject.ratings.securityInput.securityInputNeeded === 'Yes') {
+      const currentCategorisation = await getCategorisationRecord(bookingId)
+      const currentStatus = currentCategorisation.status
+      const status = Status[currentStatus]
+      if (Status.SECURITY_MANUAL.previous.includes(status) && currentStatus !== Status.SECURITY_MANUAL) {
```

Comment:
should we have the state validation in one place with a generic state transition error message?